### PR TITLE
codeintel: Rewrite janitor

### PR DESCRIPTION
**Pull request by efritz**
This is an attempt to clean up the janitor to better fit Postgres as the data store. This is partially a cleanup effort from the migration (RFC 235). There was a lot of cruft around this piece. The janitor now has two distinct responsibilities, some of which will be permanent:

- Remove all upload files on disk older than a certain age
- Remove all upload part files on disk older than a certain age
- Soft delete lsif_uploads records that are uploading longer than a given time period
- Soft delete lsif_uploads records with no matching repository
- Soft delete lsif_uploads records that are not visible at tip and are older than a given time period
- Hard delete lsif_uploads records transactionally with the data associated with it in the codeintel db
- (temporary) Find all data in the codeintel db that has no matching lsif_uploads record and delete it. This is to catch stuff that falls through the cracks and should be subsumed by the task above in normal operation.

The code for the first two can be taken out entirely if we use S3/GCS with object lifecycle management and are the only _file_ management tasks that remain in code.

This started as an effort to update our monitoring for the bundle manager, but I pulled on too much of the noodle and this meatball fell out.

5 second rule for reviewers starts NOW.

**To review**: Read the new version of enterprise/cmd/precise-code-intel-bundle-manager/internal/janitor/janitor.go and compare to the list above.

This closes https://github.com/sourcegraph/sourcegraph/issues/12168 and covers any outstanding details of [RFC 205](https://docs.google.com/document/d/1kwmhGIj9Vi_v9AQYEHjKd2Ok-Lk_e4FKc_cNs88CGK8).

### CODE REVIEW

**Comment by nicksnyder** [comment]
@efritz The @sourcegraph/code-intelligence codenotify mention is not working because the team's name is @sourcegraph/code-intel. Can you update the necessary CODENOTIFY files with the correct team name?

Alternatively, you should maybe discuss with the team whether they do in fact want to be notified of every change (maybe they don't?).

File: monitoring/CODENOTIFY [hunk 1]
```diff
@@ -2,3 +2,7 @@
 
 **/* @bobheadxi
 **/* @slimsag
+
+frontend.go @efritz
+precise_code_intel_* @efritz
+precise_code_intel_* @sourcegraph/code-intelligence
```
Comment (nicksnyder) [inline monitoring/CODENOTIFY:8]: (or rename the team to @sourcegraph/code-intelligence)
```suggestion
precise_code_intel_* @sourcegraph/code-intel
```

Comment (efritz) [reply monitoring/CODENOTIFY:8]: https://github.com/sourcegraph/sourcegraph/pull/14696

**Review by Strum355 — APPROVED** [review]
nothing stands out as wrong here :) I see https://github.com/sourcegraph/sourcegraph/pull/14695/ makes some changes to similar files, any conflicts?

File: enterprise/cmd/precise-code-intel-bundle-manager/internal/janitor/janitor.go [hunk 2]
```diff
@@ -29,72 +31,253 @@ func New(
...
+		})
+		if err != nil {
+			j.error("Failed to get deleted upload identifiers", "error", err)
+			return
+		}
+
+		var ids []int
+		for _, upload := range uploads {
+			ids = append(ids, upload.ID)
+		}
```
Comment (Strum355) [inline enterprise/cmd/precise-code-intel-bundle-manager/internal/janitor/janitor.go:185]: Just on a really pedantic note I do like preallocating if size is known, but I dont assume this to be particularly high traffic so its w/e

**Comment by efritz** [comment]
@Strum355 Nope, #14695 is stacked on top of this branch.

File: enterprise/cmd/precise-code-intel-bundle-manager/internal/janitor/janitor.go (hunk 2 above)
Comment (efritz) [reply enterprise/cmd/precise-code-intel-bundle-manager/internal/janitor/janitor.go]: Updated.

**Pull request merged**